boto/utils.py

```diff
@@ -859,17 +862,35 @@ def notify(subject, body=None, html_body=None, to_string=None,
             boto.log.exception('notify failed')
 
 
-def get_utf8_value(value):
-    if isinstance(value, bytes):
-        value.decode('utf-8')
-        return value
-    if not isinstance(value, six.string_types):
-        value = six.text_type(value)
+def _TryEncodings(string, conv_method):
+    new_string = None
+    encodings = ["utf-8", "latin-1", "ascii", "cp1252"]
 
-    if isinstance(value, six.text_type):
-        value = value.encode('utf-8')
+    def _try_enc(string, encoding):
```

Comment:
Two nits: This method might benefit from a docstring explaining what it does, and maybe consider a name for `conv_method` that describes what kind of function it should be handling.

Reply:
Good point! Docstrings added.